test/integration-test-main.sh

```diff
@@ -2255,59 +2263,58 @@ function test_not_existed_dir_obj() {
     echo data2 | aws_cli s3 cp - "s3://${TEST_BUCKET_1}/${OBJECT_NAME_2}"
 
     # Top directory
-    # shellcheck disable=SC2010
-    if ! ls -1 | grep -q '^not_existed_dir_single$'; then
-    echo "Expect to find \"not_existed_dir_single\" directory, but it is not found"
-    return 1;
+    local _LISTUP_OBJNAMES; _LISTUP_OBJNAMES="$(ls -1)"
```

Comment:
You don't need `ls -1` here -- since the terminal is non-interactive `ls` will be fine.